### ISSUE TITLE
Battle problems with npcs -cannot battle against npc's, only wild battle

### ISSUE THREAD

**Issue by Qiangong2** [Tuxemon/Tuxemon#290]
start_battle does nothing with the npcs. No error or anything, just nothing happens. I've uploaded it in my code.

**Comment by ShadowApex** [comment]
Hey @Qiangong2 . Thanks for opening this issue. I'll try and look into testing NPC battles with the latest development branch when I can.

**Comment by Qiangong2** [comment]
Update: start_battle does work, but only as a wild battle. It will follow the JSON file, but it will not say "(blank) has challenged you to a battle"

**Comment by bitcraft** [comment]
This is a good point to plan out how the start_battle command should work.  @ShadowApex may have some insight.  As is, there is no code to open a dialog.

@Qiangong2 what do you think start_battle should do?  We could modify it, or make a new command for what you are doing.

**Comment by Qiangong2** [comment]
start_battle right now works for wild, random encounters. Maybe npc_battle? With an "animation id" in the corresponding npc JSON file. That way, it would be able to discern whether it is a "Tuxemon breeder" or "Hiker" or "Tuxemon Trainer" and the corresponding person would slide in from the side and then slide out again. Like how the player shows up during the battle at the beginning. The npc would do the same, but from the other side of the screen.

**Comment by bitcraft** [comment]
Thanks for your comments.  I think that right now, start_battle is supposed to be for NPC's, but it just doesn't have the correct animation for trainers; this could be an issue with the combat state, or the action itself.  I've never tested battle with another trainer.

Next week, I will investigate other trainers in the battle.  For now, you can use start_battle.  Eventually, it will support a trainers and his monsters on the other side.  During testing, you can just pretend it works correctly.  :D

**Comment by Qiangong2** [comment]
Okay, I will :D

**Comment by Qiangong2** [comment]
@ShadowApex @bitcraft Well, its been six months...

Any update on this? It's not necessarily pressing at the moment, just an annoyance.

**Comment by bitcraft** [comment]
Its not a priority for me at the moment, but I will get to it sometime.

**Comment by Sanglorian** [comment]
Just adding to this - as well as opening dialogue and the trainer sprite/animation, it would be good to have dialogue at the end of the battle too - with different results for whether the protagonist won or lost the battle.

**Comment by xirsoi** [comment]
What's the issue here? It's possible to battle a trainer already, though the experience needs improvement.

**Comment by Qiangong2** [comment]
#613 and #604 need to be addressed first before we can close

**Comment by xirsoi** [comment]
I... disagree. Those are separate polishing issues, not bugs preventing it from working, which is what this ticket was about.

**Comment by bitcraft** [comment]
Trainer battles are working but are missing some key features.  I think this is safe to close.

**Comment by Qiangong2** [comment]
Closing then
